nntrainer/models/neuralnet.cpp

```
@@ -472,9 +499,12 @@ sharedConstTensors NeuralNetwork::inference(sharedConstTensors X) {
     setBatchSize(X[0]->batch());
   }
 
+  sharedConstTensors out;
+  if (!validateInput(X))
+    return out;
```

how about throw instead of return out?

---

Line 516 follows the same behavior of return out with a log.
So does train().
I think let's leave it with the same behavior for now and later change all to throw exceptions on error.

---

Sounds decent